abis_mapping/plugins/list.py

```diff
@@ -0,0 +1,103 @@
+"""Provides custom frictionless list plugin for the package"""
+
+
+# Third-Party
+import frictionless
+
+# Typing
+from typing import Any, Optional
+
+
+class ListPlugin(frictionless.Plugin):
+    """Custom List Plugin"""
+
+    # Class Attributes
+    code = "list"
+    status = "stable"
+
+    def create_type(self, field: frictionless.Field) -> Optional[frictionless.Type]:
+        """Create type from this plugin
+
+        Args:
+            field (frictionless.Field): Corresponding field.
+
+        Returns:
+            Optional[frictionless.Type]: Possible type from this plugin.
+        """
+        # Check for our type
+        if field.type == ListType.code:
+            # Return
+            return ListType(field)
+
+        # Not our type
+        return None
+
+
+class ListType(frictionless.Type):
+    """Custom List Type Implementation."""
+
+    # Class Attributes
+    code = "list"
+    builtin = False
+    constraints = [
+        "required",
+        "minLength",
+        "maxLength",
+        # "pattern",  # TODO -> Check whether this works
+        # "enum",  # TODO -> Check whether this works
+    ]
+
+    # Delimiter
+    delimiter = " "
+
+    def read_cell(self, cell: Any) -> Optional[list[str]]:
+        """Convert cell (read direction)
+
+        Args:
+            cell (Any): Cell to convert
+
+        Returns:
+            Optional[list[str]]: Converted cell, or none if invalid.
+        """
+        # Check that cell is not already a "list"
+        if not isinstance(cell, list):
```

Comment:
is this mean that if comma `,` is used as a delimiter frictionless will parse it as a `list`?

Reply:
This just means if you construct a cell programmatically you can pass it a `list[T]` without it doing any of the parsing stuff.

This is how its currently done in `frictionless` internally. E.g., for their inbuilt `datetime` type it checks whether the passed in object is a datetime already.

See: https://github.com/frictionlessdata/frictionless-py/blob/main/frictionless/types/datetime.py#L28